### PR TITLE
Serialize the entire trace flags not only the sample bit

### PR DESCRIPTION
**Pull request by bogdandrutu**
Signed-off-by: Bogdan Drutu <bogdandrutu@gmail.com>

### CODE REVIEW

**Review by jkwatson — COMMENTED** [review]
Can we ave a unit test for this change?

**Comment by jkwatson** [comment]
> Can we have a unit test for this change?

ping @bogdandrutu

**Comment by bogdandrutu** [comment]
@jkwatson PTAL

File: api/src/test/java/io/opentelemetry/trace/propagation/HttpTraceContextTest.java [hunk 1]
```diff
@@ -177,6 +177,17 @@ void extract_NullCarrier() {
                 TRACE_ID_BASE16, SPAN_ID_BASE16, SAMPLED_TRACE_OPTIONS, TRACE_STATE_DEFAULT));
   }
 
+  @Test
+  void extractAndInject_MoreFlags() {
+    String traceParent = "00-" + TRACE_ID_BASE16 + "-" + SPAN_ID_BASE16 + "-03";
+    Map<String, String> extractCarrier = new LinkedHashMap<>();
+    extractCarrier.put(TRACE_PARENT, traceParent);
+    Context context = httpTraceContext.extract(Context.current(), extractCarrier, getter);
+    Map<String, String> injestCarrier = new LinkedHashMap<>();
```
Comment (jkwatson) [inline api/src/test/java/io/opentelemetry/trace/propagation/HttpTraceContextTest.java:186]: tiny spelling nit: this should be `ingestCarrier`.

Comment (jkwatson) [reply api/src/test/java/io/opentelemetry/trace/propagation/HttpTraceContextTest.java:186]: or `injectCarrier` I guess. either way. :)

**Review by jkwatson — APPROVED** [review]
Thanks!

**Review by carlosalberto — APPROVED** [review]

File: api/src/test/java/io/opentelemetry/trace/propagation/HttpTraceContextTest.java [hunk 2]
```diff
@@ -177,6 +177,17 @@ void extract_NullCarrier() {
...
+    httpTraceContext.inject(context, injestCarrier, setter);
+    assertThat(extractCarrier).isEqualTo(injestCarrier);
```
Comment (anuraaga) [inline api/src/test/java/io/opentelemetry/trace/propagation/HttpTraceContextTest.java:188]:
```suggestion
    Map<String, String> injectCarrier = new LinkedHashMap<>();
    httpTraceContext.inject(context, injectCarrier, setter);
    assertThat(extractCarrier).isEqualTo(injectCarrier);
```

**Review by anuraaga — APPROVED** [review]
Thanks!

**Pull request merged**